### PR TITLE
Add gas estimation in self execute function [After #2373]

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-commons/src/core/constants/gas.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const DEFAULT_GAS_LIMIT_EXECUTION = 80000;
 
 export const DEPLOYMENT_REFUND = utils.bigNumberify(570000);
 
-export const GAS_BASE = 30000;
+export const GAS_BASE = 20000;
```
Comment (xitronix) [inline universal-login-commons/src/core/constants/gas.ts:15]: Why do we decrease the value?

File: universal-login-sdk/src/api/wallet/DeployedWallet.ts [hunk 2]
```diff
@@ -28,16 +29,20 @@ export class DeployedWallet extends AbstractWallet {
     };
   }
 
-  addKey(publicKey: string, executionOptions: ExecutionOptions): Promise<Execution> {
-    return this.selfExecute('addKey', [publicKey], {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
+  async addKey(publicKey: string, executionOptions: ExecutionOptions): Promise<Execution> {
```
Comment (xitronix) [inline universal-login-sdk/src/api/wallet/DeployedWallet.ts:32]: Should we have async here?

File: universal-login-sdk/src/api/wallet/DeployedWallet.ts [hunk 3]
```diff
@@ -51,11 +56,7 @@ export class DeployedWallet extends AbstractWallet {
     await this.sdk.relayerApi.denyConnection(authorisationRequest);
   }
 
-  setRequiredSignatures(requiredSignatures: number, executionOptions: ExecutionOptions): Promise<Execution> {
-    return this.selfExecute('setRequiredSignatures', [requiredSignatures], {gasLimit: DEFAULT_GAS_LIMIT, ...executionOptions});
-  }
-
-  async execute(message: Partial<Message>): Promise<Execution> {
+  async getSignedMessage(message: Partial<Message>) {
```
Comment (xitronix) [inline universal-login-sdk/src/api/wallet/DeployedWallet.ts:59]: Should it be private?

**Comment by JustynaBroniszewska** [comment]
Will be done in different pr

**Pull request closed without merging**